src/sage/rings/polynomial/polynomial_modn_dense_ntl.pyx

```diff
@@ -1799,6 +1789,49 @@ cdef class Polynomial_dense_mod_p(Polynomial_dense_mod_n):
     A dense polynomial over the integers modulo p, where p is prime.
     """
 
+    def __pow__(self, n, modulus):
+        """
+        TESTS::
+
+            sage: R.<x> = PolynomialRing(Integers(101), implementation='NTL')
+            sage: (x-1)^5
+            x^5 + 96*x^4 + 10*x^3 + 91*x^2 + 5*x + 100
+            sage: pow(x-1, 15, x^3+x+1)
+            55*x^2 + 6*x + 46
+
+        Negative powers will not work::
+
+            sage: R.<x> = PolynomialRing(Integers(101), implementation='NTL')
+            sage: (x-1)^(-5)
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: Fraction fields not implemented for this type.
+
+        We define ``0^0`` to be unity, :trac:`13895`::
+
+            sage: R.<x> = PolynomialRing(Integers(101), implementation='NTL')
+            sage: R(0)^0
+            1
+
+        The value returned from ``0^0`` should belong to our ring::
+
+            sage: R.<x> = PolynomialRing(Integers(101), implementation='NTL')
+            sage: type(R(0)^0) == type(R(0))
+            True
+        """
+        if modulus is not None and not isinstance(modulus, Polynomial_dense_mod_n):
+            return Polynomial.__pow__(self, n, modulus)
+
+        if modulus is not None:
+            return self.parent()(pow(self.ntl_ZZ_pX(), n, modulus.ntl_ZZ_pX()), construct=True)
+        else:
+            if n < 0:
+                return ~(self**(-n))
+            elif self.degree() <= 0:
+                return self.parent()(self[0]**n)
+            else:
+                return self.parent()(self.ntl_ZZ_pX()**n, construct=True)
```

Comment:
no it doesn't work

Reply:
It fails at runtime saying that there is no `_parent` attribute. I think `_parent` is a C field and here it would try to access a Python field ? Does it need a specific syntax ? (some parts of the code say `(<Element>self)._parent`).

Reply:
Indeed, it works after casting to `Element`, updated in pull request.

Reply:
Thanks for checking and changing it.